docs/ldml/tr35-keyboards.md

```diff
@@ -308,6 +308,77 @@ For extensibility, the `<special>` element will be allowed at nearly every level
 
 See [Element special](tr35.md#special) in Part 1.
 
+## Normalization
+
+Normalization will not typically be the responsibility of the keyboard author, rather this will be managed by the implementation.
+The implementation will apply normalization as appropriate when matching transform rules and `<display>` value matching.
+Output from the keyboard, following application of all transform rules, will be normalized to implementation- or application-requested form.
+
+The attribute value `normalization="disabled"` can be used to indicate that no automatic normalization happens.
+Using this setting is not recommended. See [`<settings>`](#element-settings) for details.
+
+Input source files may be in any normalization format, however, authors should be aware of two areas where normalization affects keyboard operation: that of transform matching, and that of output.
+
+Normalization can be applied within segments defined by Grapheme Cluster Boundaries as defined by [UAX #29](https://www.unicode.org/reports/tr29/#Grapheme_Cluster_Boundaries). This segmentation is pertinent where markers are involved, see below.
```

Comment:
What does this mean? That it _must_ be applied within UAX #29 boundaries? Or a warning that not whole string might be normalized at once? What if implementations employ different clustering rules?

Reply:
Perhaps this isn't helpful here and should move down to the discussion of markers. All it means is that the marker interaction doesn't occur across base char boundaries.

Reply:
I agree it doesn't have much value here, and that it would be preferable if we had a simpler rule about the boundary, perhaps based on the character properties only, such as non-mark characters.

Reply:
Based on what we said in the meeting:

> Normalization will only applied within normalization-safe boundary distance from the caret, within segments as defined by...

To make it clear that using an LDML keyboard doesn't instantly normalize the whole doc, a clarification we noted.

Reply:
yes, i will move that comment under the discussion of output. And let the marker discusion stand on its own.